docs/source/nlp/joint_intent_slot.rst

```diff
@@ -3,10 +3,253 @@
+- :code:`train.tsv/test.tsv` - A list of original queries, one per line, with the intent number
+  separated by a tab (e.g. "what alarms do i have set right now <TAB> 0"). Intent numbers are
+  set according to the intent line in the intent dictionary file (:code:`dict.intents.csv`),
+  starting from 0. The first line in these files should contain the header line "sentence
+  <tab> label".
+- :code:`train_slot.tvs/test_slot.tsv` - A list that contains one line per a query, when an each word from the original text queries,
+  is replaced by a token number from the slots dictionary file (dict.slots.csv), counted starting from 0.
+  All the words which do not contain a relevant slot are replaced by 'out-of scope' token number, which is also a part of the slot dictionary file,
+  usually as the last entry there. Example of a line from these files: "54 0 0 54 54 12 12" (the numbers are separated by a space).
+  These files do not contain any header line.
+
+
+Dataset Conversion
+------------------
+
+To convert to the format of the model data, use the `import_datasets` utility, which implements
+conversion for the Assistant dataset, which you can download `here <https://github.com/xliuhw/NLU-Evaluation-Data>`_.
+Or you can write your own converter for the format that you are using for data annotation.
+
+For a dataset that follows your own annotation format, we recommend using one text file for all
+samples of the same intent, with the name of the file as the name of the intent. Use one line per
+query, with brackets to define slot names. This is very similar to the assistant format, and you can
+adapt this converter utility or your own format with small changes:
+
+::
+
+    did i set an alarm to [alarm_type : wake up] in the [timeofday : morning]
+
+To run the dataset_converter, use this command:
+
+.. code::
+
+    python examples/nlp/intent_slot_classification/data/import_datasets.py
+        --source_data_dir=`source_data_dir` \
+        --target_data_dir=`target_data_dir` \
+        --dataset_name=['assistant'|'snips'|'atis']
+
+- :code:`source_data_dir`: The directory location of the your dataset
+- :code:`target_data_dir`: The directory location where the converted dataset should be saved
+- :code:`dataset_name`: One of the implemented dataset names
+
+After conversion, the `target_data_dir` should contain the following files:
+
+.. code::
+
+   .
+   |--target_data_dir
+     |-- dict.intents.csv
+     |-- dict.slots.csv
+     |-- train.tsv
+     |-- train_slots.tsv
+     |-- test.tsv
+     |-- test_slots.tsv
+
+
+
+Model Training
+--------------
+
+This is a pretrained Bert based model with 2 linear classifier heads on the top of it,
+one for classifying an intent of the query and another for classifying slots for each token of the query.
+This model is trained with the combined loss function on the Intent and Slot classification task on the given dataset.
+The model architecture is based on the paper: https://arxiv.org/pdf/1902.10909.pdf
```

Comment:
Add link to the paper + reference to the paper

Reply:
Done